src/inc/volatile.h

```diff
@@ -479,7 +479,7 @@ class VolatilePtr : public Volatile<P>
 #else
 
 // Disable use of Volatile<T> for GC/HandleTable code except on platforms where it's absolutely necessary.
-#if defined(_MSC_VER) && !defined(_ARM_)
+#if defined(_MSC_VER) && !defined(_ARM_) && !defined(_ARM64_)
```

Comment:
This issue has been causing frequent deadlocks under server GC, all of which were related to the GC's t_join algorithm breaking down due to writes to volatile join_structure fields being observed out of order on remote threads.